1.49.0: build fails on NetBSD

<!--
If you want to report a bug, you are in the right place!

If you need help or have a question, go here:
https://github.com/libuv/libuv/discussions

If you are reporting a libuv test failure, please ensure that you are not
running the test as root.

Please include code that demonstrates the bug and keep it short and simple.
-->
* **Version**: 1.49.0
* **Platform**: NetBSD 10.99.12

Building 1.49.0 fails on NetBSD with (1.48.0 was fine):

```
  CC       src/unix/libuv_la-core.lo
In file included from /usr/include/sched.h:37,
                 from /usr/include/pthread.h:38,
                 from ./include/uv/unix.h:43,
                 from ./include/uv.h:71,
                 from src/unix/core.c:21:
src/unix/core.c: In function 'uv__cpu_count':
src/unix/core.c:1891:26: error: 'cpu' undeclared (first use in this function)
 1891 |     int r = cpuset_isset(cpu, set);
      |                          ^~~
src/unix/core.c:1891:26: note: each undeclared identifier is reported only once for each function it appears in
src/unix/core.c:1891:31: error: 'set' undeclared (first use in this function); did you mean 'sem_t'?
 1891 |     int r = cpuset_isset(cpu, set);
      |                               ^~~
src/unix/core.c: In function 'uv_available_parallelism':
src/unix/core.c:1932:58: warning: passing argument 3 of 'sched_getaffinity_np' from incompatible pointer type [-Wincompatible-pointer-types]
 1932 |     if (0 == sched_getaffinity_np(getpid(), sizeof(set), &set))
      |                                                          ^~~~
      |                                                          |
      |                                                          cpuset_t ** {aka struct _cpuset **}
/usr/include/sched.h:67:45: note: expected 'cpuset_t *' {aka 'struct _cpuset *'} but argument is of type 'cpuset_t **' {aka 'struct _cpuset **'}
   67 | int     sched_getaffinity_np(pid_t, size_t, cpuset_t *);
      |                                             ^~~~~~~~~~
src/unix/core.c:1933:26: warning: passing argument 1 of 'uv__cpu_count' from incompatible pointer type [-Wincompatible-pointer-types]
 1933 |       rc = uv__cpu_count(&set);
      |                          ^~~~
      |                          |
      |                          cpuset_t ** {aka struct _cpuset **}
src/unix/core.c:1885:36: note: expected 'cpuset_t *' {aka 'struct _cpuset *'} but argument is of type 'cpuset_t **' {aka 'struct _cpuset **'}
 1885 | static int uv__cpu_count(cpuset_t *cpuset) {
      |                          ~~~~~~~~~~^~~~~~
```

Here's my patch to fix it:

```
--- src/unix/core.c.orig	2024-09-25 08:17:20.000000000 +0000
+++ src/unix/core.c
@@ -1882,13 +1895,13 @@ int uv__search_path(const char* prog, ch
 #if defined(__linux__) || defined (__FreeBSD__)
 # define uv__cpu_count(cpuset) CPU_COUNT(cpuset)
 #elif defined(__NetBSD__)
-static int uv__cpu_count(cpuset_t *cpuset) {
+static int uv__cpu_count(cpuset_t *set) {
   int rc;
   cpuid_t i;
 
   rc = 0;
   for (i = 0;; i++) {
-    int r = cpuset_isset(cpu, set);
+    int r = cpuset_isset(i, set);
     if (r < 0)
       break;
     if (r)
```

--- a/src/unix/core.c
+++ b/src/unix/core.c
@@ -1882,13 +1882,13 @@ int uv__search_path(const char* prog, char* buf, size_t* buflen) {
 #if defined(__linux__) || defined (__FreeBSD__)
 # define uv__cpu_count(cpuset) CPU_COUNT(cpuset)
 #elif defined(__NetBSD__)
-static int uv__cpu_count(cpuset_t *cpuset) {
+static int uv__cpu_count(cpuset_t *set) {
   int rc;
   cpuid_t i;
 
   rc = 0;
   for (i = 0;; i++) {
-    int r = cpuset_isset(cpu, set);
+    int r = cpuset_isset(i, set);
     if (r < 0)
       break;
     if (r)

--- a/src/unix/core.c
+++ b/src/unix/core.c
@@ -1882,7 +1882,7 @@ int uv__search_path(const char* prog, char* buf, size_t* buflen) {
 #if defined(__linux__) || defined (__FreeBSD__)
 # define uv__cpu_count(cpuset) CPU_COUNT(cpuset)
 #elif defined(__NetBSD__)
-static int uv__cpu_count(cpuset_t *set) {
+static int uv__cpu_count(cpuset_t* set) {
   int rc;
   cpuid_t i;
 

